QuickOSM/test/test_overpass_panel.py

```diff
@@ -13,40 +17,68 @@
 
 class TestQuickOSMWidget(unittest.TestCase):
 
+    def setUp(self):
+        self.dialog = Dialog(get_iface())
+        layer_path = plugin_test_data_path('layer', 'layertest.qpkg')
```

Comment:
layertest.qpkg ? Not **g**pkg ?

I would suggest you to use a simple QgsVectorLayer with "memory" provider.
https://docs.qgis.org/testing/en/docs/pyqgis_developer_cookbook/vector.html#from-an-instance-of-qgsvectorlayer
Not need to have a local file.

I prefer to use text-based data, easier to debug, to follow on git etc.
GPKG is a binary format, we can't make "git diff" on the file.
If you need a layer, you can always use a "geojson" layer.

Reply:
I tested with a geojson and during the test I triggered "qgis died in signal 11" (same with shp)